main.cpp

```diff
@@ -283,529 +195,165 @@ struct Numeric
                     return *this;
                 }
             }
-            else if (rhs < std::numeric_limits<OtherType>::epsilon())
+            else if (static_cast<OtherType>(rhs) < std::numeric_limits<OtherType>::epsilon())
```

Comment:
`rhs` is already an `OtherType`.  
This cast is redundant
Don't revise code that doesn't need revising
- [ ] revert